Automattic/wp-calypso
# Jetpack Agency Dashboard: Add Filter UI

#### Changes proposed in this Pull Request

This PR adds a filter to the agency dashboard for filtering down to sites that have a particular issue.

<img width="1344" alt="image" src="https://user-images.githubusercontent.com/42627630/171312568-8f2096a5-300c-4024-be0d-c8698e697ca0.png">

<img width="293" alt="image" src="https://user-images.githubusercontent.com/42627630/171312649-5d1e195b-fb58-4a9f-9cdd-0d07e8c74891.png">

<img width="1340" alt="image" src="https://user-images.githubusercontent.com/42627630/171312688-c6ac7f43-9917-461c-8c54-9466fc419df6.png">

To develop this PR I re-used the filter which is found at `client/my-sites/activity/filterbar`. This took a bit of refactoring, particularly factoring out the components which are found at `client/my-sites/activity/filterbar/type-selector` and making modifications to the main `Filterbar` component. I don't think this component is quite ready for generic use so I haven't moved it to a generic components folder. With some further refinement I think this component could be made ready for more generic use, but that would ideally be its own PR.


#### Testing instructions



Note: if the checkboxes do not show a checkmark when you select a filter option then check the console for 404 errors coming from GET requests to /calypso/images/checkbox-icons/checkmark-primary.svg. This also happens on `trunk` so it appears to be an issue in the dev environment which isn't happening in the production environments.

1. Visit `http://jetpack.cloud.localhost:3001/dashboard`. Select various filters and verify that the UI updates appropriately. Note that the logic currently ANDs the filter options while OR logic would be more appropriate, but that logic is done server side.
2. Verify that you can clear the issues after making a selection by clicking the X on the filterbar.
3. Visit `http://jetpack.cloud.localhost:3001/activity-log/:site_slug` and verify that that nothing has changed with the "Activity type" filter. You should still be able to select and clear all of the filters.



Related to 1202076982646589-as-1202180914172733
